### ISSUE TITLE
Add environment variable for configuring labels set on a flow during registration

### ISSUE THREAD

**Issue by ciaranevans** [PrefectHQ/prefect#4294]
## Current behavior

Hey folks,

Currently, the Agent syntax for providing labels via an environment variable is:

```
export PREFECT__CLOUD__AGENT__LABELS='["dev", "staging"]'
```

Because of this, when deploying my Agents via AWS CDK, I can just provide this list like:

```python
environment={
    "PREFECT__CLOUD__AGENT__LABELS": os.environ["PREFECT_AGENT_LABELS"]
},
```

Where `PREFECT_AGENT_LABELS` is an environment variable in the form `'["dev", "staging"]'` 🎉 

Currently (as of writing with version `0.14.13`) labels can only be applied to a Flow singularly with the `--label` value in the CLI. It'd be really nice if this worked in the same manner as `PREFECT__CLOUD__AGENT__LABELS`

## Proposed behavior

I'd expect a scenario where I would run:

```
$ prefect register --project AProject --path flow.py --name "hello-flow" --labels '["dev", "staging"]'
```

## Example

This would be mainly useful in A: having consistency on how Labels are applied and B: I could use the same environment variable when deploying my Agents AND when registering my flows for those Agents.

In other words, I have a `.env` file with `PREFECT_AGENT_LABELS` in, with the list structure and then I can reference that in CDK and my CLI call.

Appreciate all your work!

**Comment by jcrist** [comment]
The agent CLI also has a `--label` flag, so really the environment variable you're using is the odd case :). You can set labels on the agent with the same syntax as `prefect register`:

```shell
$ prefect agent local start --label dev --label staging
```

If you were to have an environment variable for configuring flow labels, how would you expect it to behave? Right now a flow's labels are the combined set of:

- Things set explicitly on the flow
- Extra labels added via the `--label` flag to the CLI

Would the environment variable be always used, for the combined set of all three options? Would it only be used if one of the options was unset (a default), and if so, which one? Currently the agent environment variable is only used if no explicit `--label` flag is passed, so parity with that would be a default for the CLI option if no `--label` flag is used.

**Comment by ciaranevans** [comment]
Hmmm I'm honestly not too sure.

Ultimately being able to pass a list is easier to do automatically than have to get some kind of parsing to do multiple `--label` statements IMO. 

For reference, my Agent is being started _without_ the `--label` flag as I saw the Environment variable option and that's a lot more flexible in terms of a CDK deployment (I don't have to rebuild the image etc.) So the ability to use the same list to register the flow would be ace.

**Comment by ciaranevans** [comment]
I should caveat this with I'm not a prolific Prefect user _yet_ so I may be a very odd-one-out case, just posing this as something I'd find useful 😅

**Comment by madkinsz** [comment]
@ciaranevans have you considered just using `os.environ` to load the environment variable you want and pass them to `flow.register(labels=[...])`?

**Comment by ciaranevans** [comment]
@madkinsz I guess that would be one way to go about it, it'd just mean having to parse the string list in the environment variable into a list of strings.

I could very well just take off the `--label` part from my CLI call and have that in the flow definition though

**Comment by jcrist** [comment]
> I could very well just take off the `--label` part from my CLI call and have that in the flow definition though

> edit: though I'm not sure if you can set it in the flow definition 

Yeah, that'd be my suggestion for now. You'd do this by adding labels to your flow's run-config (see https://docs.prefect.io/orchestration/flow_config/run_configs.html#labels).

I'm going to relabel this issue to be more specific, and think about what api we'd like to expose here.

**Comment by ciaranevans** [comment]
Cool thanks @jcrist - I'll be honest, I hadn't looked into the `run_config` side, that's probably good enough!
